core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -62,6 +62,12 @@ class TopicConfigHandler(private val logManager: LogManager, kafkaConfig: KafkaC
     topicConfig.asScala.forKeyValue { (key, value) =>
       if (!configNamesToExclude.contains(key)) props.put(key, value)
     }
+
+    if (!kafkaConfig.remoteLogManagerConfig.enableRemoteStorageSystem()
+      && topicConfig.getProperty(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG) != null) {
```

Comment:
Why are we checking only for the existence of the property instead of checking whether that property is set as true? This requires more validation checks whenever this config value is set as true but it gives better flexibility to users. 

Is it to keep the remote storage related configs at topic level separation better?

Reply:
In summary, I thought it was just one more check of the kind "are you absolutely certain you have checked this and would like to proceed", but I do not feel too strongly about it if you think we should just fail if the value is set to true.

Reply:
I'd disagree with this. Under this change, when the users set the topic level `enable.remote.storage=false` explicitly when creating a topic. Then, when disabling system-wide `remote.log.storage.system.enable=false`, the startup will fail immediately with: 
```
s"You have to delete all topics with the property remote.storage.enable (i.e. $topic) before disabling tiered storage cluster-wide"
```
That doesn't make sense to me. 

If you think we should notify users about this:
```
Right now (and for 3.6) if I change enable.remote.storage from true to false the topic will stop being archived, but the data which is already in remote storage won't be deleted.
```
I think we can add note in docs and logs when dynamically disabled topic-level `enable.remote.storage=false`, and make this check to `true` case.  WDYT?

Reply:
Sure, I will change to condition to only check for the value being true rather than its presence! As mentioned, I am happy with either 😊